### PR TITLE
Improve search

### PR DESCRIPTION
**Pull request by BohdanBusko**
**STATUS**
READY

**DESCRIPTION**
screenshot from 2018-11-15 17-11-30
screenshot from 2018-11-15 17-29-55
screenshot from 2018-11-15 17-29-49

**CHECK LIST**
- [ ] specs added
- [ ] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs

### CODE REVIEW

File: app/views/account/tasks/new.html.slim [hunk 1]
```diff
@@ -1,4 +1,5 @@
-- title "New Task"
+- title params[:section].nil? ? 'New Task' : 'New Section'
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/new.html.slim]: DRY CODE .. I see same code 2 times

File: app/views/account/tasks/new.html.slim [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
 .container.box
-  .title New Task
+  .title
+    = params[:section].nil? ? 'New Task' : 'New Section'
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/new.html.slim]: what about form fields? ALL OF THEM APPLICABLE FOR SECTION?>????? THINK!!!!!!!

File: app/assets/javascripts/search.js [hunk 3]
```diff
@@ -30,6 +31,12 @@ function globalSearch() {
             display(response.workspaces, workspacesBlock, workspaces, 'WORKSPACES', 'name');
             display(response.projects, projectsBlock, projects, 'PROJECTS', 'name');
             display(response.tasks, tasksBlock, tasks, 'TASKS', 'title');
+
+            if(response.workspaces.length <= 0 && response.projects.length <= 0 && response.tasks.length <= 0) {
```
Comment (igorkasyanchuk) [inline app/assets/javascripts/search.js:35]: add screenshot how page look when no results

File: app/views/account/tasks/_form.html.slim [hunk 4]
```diff
@@ -1,8 +1,9 @@
 = simple_form_for [:account, @project, @task] do |f|
   = f.input_field :section, as: :hidden, value: @task.section
   = f.input :title, placeholder: 'Title', wrapper: :bulma_input, required: true
-  = f.input :description, placeholder: 'Description', wrapper: :textarea
-  = f.input :due_date, wrapper: :date, as: :date, html5: true
-  = f.input :files, input_html: { multiple: true }, wrapper: :file, label: false
+  - if params[:section].nil?
```
Comment (igorkasyanchuk) [inline app/views/account/tasks/_form.html.slim]: коли я редагую секшен - то що в параметрах?
як ці поля нижче будуть приховані?

File: app/assets/javascripts/search.js [hunk 5]
```diff
@@ -14,6 +14,7 @@ function globalSearch() {
   var workspacesBlock = document.getElementById('workspaces-block');
   var projectsBlock = document.getElementById('projects-block');
   var tasksBlock = document.getElementById('tasks-block');
+  var noContentBlock = document.getElementById('no-content-block');
```
Comment (igorkasyanchuk) [inline app/assets/javascripts/search.js:17]: коли нема результатів показуй No results found

File: app/helpers/account/tasks_helper.rb [hunk 6]
```diff
@@ -8,4 +8,8 @@ def task_link_class(task)
       'has-text-link'
     end
   end
+
+  def section_or_task
```
Comment (igorkasyanchuk) [inline app/helpers/account/tasks_helper.rb]: find better name for method

File: app/views/account/projects/_show_tasks.html.slim [hunk 7]
```diff
@@ -24,7 +25,7 @@
         p
           | Created  #{time_ago_in_words(task.created_at)} ago
       .level-item
-        = link_to "Edit", edit_account_project_task_path(project, task),
+        = link_to "Edit", edit_account_project_task_path(project, task, ({section: task.section} if task.section?)),
```
Comment (igorkasyanchuk) [inline app/views/account/projects/_show_tasks.html.slim]: THIS IS BAD

File: app/views/account/tasks/_form.html.slim (hunk 4 above)
Comment (BohdanBusko) [reply app/views/account/tasks/_form.html.slim]: так вони будуть приховані

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**